src/ddmd/dsymbol.d

```diff
@@ -143,31 +143,14 @@ struct Prot
     }
 }
 
-enum PASS : int
+enum SemState : uint
```

Comment:
This breaks downstream compilers written in c++, can't we just keep the same PASS enum, just simplify it?

Reply:
Does this break of a lot of GDC code?

semantic1/2/3() are getting split into smaller functions, that may be called separately. For each of these functions I need to store an independent SemState value. SemState could be renamed to PASS, but won't the rest of the changes be heavy for GDC anyway?

Reply:
PASSobj is used to determine if code has been generated for the declaration.  And semantic may be requested in the declaration symbol pass.

Keeping the naming convention would be a noop for downstream, just adjusting the enum as necessary to reflect D definitions.

You mention that each semantic has an independent SemState, how can you determine from the outside which stage a given symbol is currently at?